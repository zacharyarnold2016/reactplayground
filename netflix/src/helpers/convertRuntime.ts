export const convertRuntime = (runtime: number) => {
  const hours: number = Math.round((runtime as number) / 60);
  const minutes: number = (runtime as number) % 60;
  const time: string = `${hours}h${minutes}min`;
  return time;
};
