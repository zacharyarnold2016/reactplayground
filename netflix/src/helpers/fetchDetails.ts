export const fetchDetails = async (state: any) => {
  if (state.movieId !== undefined) {
    const details = await fetch(
      `http://localhost:4000/movies/${state.movieId}`
    );
    const jsonDetails = await details.json();
    console.log(jsonDetails);
    state.setDetails(jsonDetails);
  } else {
    state.setDetails(undefined);
  }
};
