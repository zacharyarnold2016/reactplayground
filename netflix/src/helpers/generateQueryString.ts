const generateQueryString = ({ searchQuery, genre, sortBy }: any) => {
  const queryParams = new URLSearchParams();
  queryParams.append("searchBy", "title");
  if (searchQuery === "f08ab2") {
    queryParams.append("search", "");
  } else if (searchQuery) {
    queryParams.append("search", searchQuery);
  }

  if (genre && genre !== "All") {
    queryParams.append("filter", genre);
  }
  sortBy && queryParams.append("sortBy", sortBy);
  sortBy && queryParams.append("sortOrder", "desc");

  const queryString = `http://localhost:4000/movies?${queryParams.toString()}`;
  return queryString;
};

export default generateQueryString;
