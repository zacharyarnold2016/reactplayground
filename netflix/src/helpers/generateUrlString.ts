import { pageObject } from "../interfaces/pageObject.interface";

const generateUrl = ({
  sortBy,
  searchQuery,
  genre = "All",
}: pageObject): string => {
  const queryParams = new URLSearchParams();
  let url: string = `/search`;
  searchQuery && (url = url + `/` + searchQuery);
  genre && queryParams.append("genre", genre as string);
  sortBy && queryParams.append("sortBy", sortBy as string);
  const returnString: string = `${url}?${queryParams.toString()}`;
  return returnString;
};

export default generateUrl;
