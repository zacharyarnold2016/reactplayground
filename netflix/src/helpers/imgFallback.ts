export const imgFallback = ({ currentTarget }: any) => {
  currentTarget.onerror = null;
  currentTarget.src = "https://picsum.photos/500/750";
};
