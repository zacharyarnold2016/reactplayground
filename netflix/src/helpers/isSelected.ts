export const isSelected = (genre: any, currentGenre: any) => {
  if (genre === currentGenre) {
    return "highlighted" as string;
  } else {
    return "" as string;
  }
};
