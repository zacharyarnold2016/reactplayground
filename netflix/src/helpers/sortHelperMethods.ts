export const sortString = (sortOption: string) => {
  const inputString = sortOption.toLowerCase();
  const array = inputString.split(" ");
  const outputString = array.join("_");
  return outputString;
};
