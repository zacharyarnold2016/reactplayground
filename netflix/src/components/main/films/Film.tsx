import React from "react";
import { useRouter } from "next/router";
import Link from "next/link";
import generateUrl from "../../../helpers/generateUrlString";
import { imgFallback } from "../../../helpers/imgFallback";

const Film = (props: any) => {
  const router = useRouter();
  const { sortBy, searchQuery, genre } = router.query;
  const url = generateUrl({ sortBy, searchQuery, genre });

  return (
    <div className="filmContainer">
      <Link href={`${url}&movieId=${props.id}`}>
        <img
          src={props.poster_path}
          className="listImage"
          alt="Failed to find Poster Path"
          onError={(currentTarget) => imgFallback(currentTarget)}
        />
      </Link>
      <div className="--details">
        <h2 className="text-red-900">{props.title}</h2>
        <h3 className="release_date">{props.release_date}</h3>
        <h3 style={{ color: "#979797" }}>{props.genres[0]}</h3>
      </div>
    </div>
  );
};

export default Film;
