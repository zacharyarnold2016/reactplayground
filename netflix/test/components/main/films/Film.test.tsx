// @ts-nocheck

import { expect } from "@jest/globals";
import renderer from "react-test-renderer";
import Film from "../../../../src/components/main/films/Film";
import { useSearchParams } from "react-router-dom";
import { imgFallback } from "../../../../src/helpers/imgFallback";

const jestSpy = jest.fn((body) => 1);
useSearchParams = jest.fn(() => {
  return [
    {
      get: (option) => option,
    },
    jestSpy,
  ];
});

imgFallback = jest.fn((value) => {
  return value;
});
describe("Film", () => {
  test("renders", () => {
    const component = renderer.create(
      <Film
        poster_path="url"
        title="test"
        release_date="testDate"
        genres={["test"]}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(useSearchParams).toBeCalled();
  });

  test("onClick", () => {
    const component = renderer.create(
      <Film
        poster_path="url"
        title="test"
        release_date="testDate"
        genres={["test"]}
      />
    );
    let tree = component.toJSON();
    tree.children[0].props.onClick();
    expect(jestSpy).toBeCalled();
  });

  test("onError", () => {
    const component = renderer.create(
      <Film
        poster_path="url"
        title="test"
        release_date="testDate"
        genres={["test"]}
      />
    );
    let tree = component.toJSON();
    tree.children[0].props.onError();
    expect(imgFallback).toBeCalled();
  });
});
