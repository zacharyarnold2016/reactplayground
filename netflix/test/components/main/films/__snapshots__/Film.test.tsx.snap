// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film renders 1`] = `
<div
  className="filmContainer"
>
  <img
    alt="Failed to find Poster Path"
    className="listImage"
    onClick={[Function]}
    onError={[Function]}
    src="url"
  />
  <div
    className="--details"
  >
    <h2>
      test
    </h2>
    <h3
      className="release_date"
    >
      testDate
    </h3>
    <h3
      style={
        Object {
          "color": "#979797",
        }
      }
    >
      test
    </h3>
  </div>
</div>
`;
